packages/flutter/test/physics/spring_simulation_test.dart

```diff
@@ -35,4 +35,56 @@ void main() {
     expect(snappingSimulation.x(time), 1);
     expect(snappingSimulation.dx(time), 0);
   });
+
+  group('SpringDescription.withDurationAndBounce', () {
+    test('creates spring with expected results', () {
```

Comment:
Add a test for custom duration for positive and negative respectively maybe?

Reply:
Done